lib/Dialect/TTIR/Analysis/GridAnalysis.cpp

```diff
@@ -11,4 +11,14 @@ void GridAnalysis::analysisImplementation() {
   analysis_result.target_rows = analysis_input.max_supported_rows;
   analysis_result.target_columns = analysis_input.max_supported_columns;
 }
+
+void GridAnalysis::handleOverride(llvm::json::Object *override) {
```

Comment:
I think instead of passing around a json object, we should deserialize the json right at the python -> c++ boundary and convert this into the respective pass options struct.

Reply:
Sure! I had done it this way because the LLVM JSON bindings are flexible enough that it could be tossed around and represent overrides of many flavours, but with an `Options` approach that functionality probably won't be needed.